cosmo_tester/test_suites/bootstrap_based_tests/distributed_installation_test.py

```diff
@@ -0,0 +1,159 @@
+        database_and_manager[0].additional_install_config = {
+            'sanity': {'skip_sanity': 'true'},
+            'postgresql_server': {'enable_remote_connections': 'true'},
+            'services_to_install': [
+                'database_service'
+            ]
+        }
+        database_and_manager[1].additional_install_config = {
+            'sanity': {'skip_sanity': 'true'},
+            'postgresql_client': {
+                'host': str(database_and_manager[0].private_ip_address)
+            },
+            'services_to_install': [
+                'queue_service',
+                'composer_service',
+                'manager_service'
+            ]
+        }
+
+    hosts = DistributedInstallationCloudifyManager(cfy=cfy,
+                                                   ssh_key=ssh_key,
+                                                   tmpdir=module_tmpdir,
+                                                   attributes=attributes,
+                                                   logger=logger)
+
+    hosts.instances[1].upload_plugins = False
+
+    hosts.preconfigure_callback = _preconfigure_callback_manager_and_database
+
+    try:
+        hosts.create()
+        yield hosts.instances
+    finally:
+        hosts.destroy()
+
+
+@pytest.fixture(scope='function')
+def nodecellar(cfy, database_and_manager, attributes, ssh_key, tmpdir, logger):
+    # Uploading to the manager not the database
+    manager = database_and_manager[1]
+    manager.use()
+    tenant = prepare_and_get_test_tenant(TENANT_NAME, manager, cfy)
+    nc = NodeCellarExample(
+        cfy, manager, attributes, ssh_key, logger, tmpdir,
+        tenant=tenant, suffix='simple')
+    nc.blueprint_file = 'simple-blueprint-with-secrets.yaml'
+    # nc.blueprint_file = 'openstack-blueprint.yaml' # alternative bp for test
+    yield nc
+
+
+def test_distributed_installation_scenario(database_and_manager,
+                                           cfy,
+                                           logger,
+                                           tmpdir,
+                                           attributes,
+                                           nodecellar):
+    manager = database_and_manager[1]
+
+    logger.info('Cfy version')
+    cfy('--version')
+
+    logger.info('Cfy status')
+    cfy.status()
+
+    _create_and_add_user_to_tenant(cfy, logger)
+
+    _set_test_user(cfy, manager, logger)
+
+    # Creating secrets
+    _create_secrets(cfy, logger, attributes, manager)
+
+    nodecellar.upload_and_verify_install()
+
+    snapshot_id = 'SNAPSHOT_ID'
+    create_snapshot(manager, snapshot_id, attributes, logger)
+
+    _set_admin_user(cfy, manager, logger)
+
+    # Restore snapshot
+    logger.info('Restoring snapshot')
+    restore_snapshot(manager, snapshot_id, cfy, logger, force=True)
+
+    # wait a while to allow the restore-snapshot post-workflow commands to run
+    time.sleep(30)
+
+    nodecellar.uninstall()
+
+
```

Comment:
Better to add the following to nodecellar example impl file as infrastructure, and remove it from other places